configmap/nodep_test.go

```diff
@@ -0,0 +1,32 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package configmap_test
+
+import (
+	"testing"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	"knative.dev/pkg/nodep"
+)
+
+func TestNoDeps(t *testing.T) {
+	nodep.AssertNoDependency(t, map[string]sets.String{
+		"knative.dev/pkg/configmap": sets.NewString(
+			"k8s.io/client-go/informers",
+		),
+	})
```

Comment:
- how does this affect test duration?
- can we avoid the package name `knative.dev/pkg/configmap` 
- do we have instances where binaries are depending on the configmap water interface but don't have informers?

Reply:
Just curious, is your feeling that exclusion list is going to be smaller, or the allow list? Just thinking about which way we think it's easier to create these tests. If we use exclusion list, then any new dependency is allowed by default. I wonder if just like we have test coverage, we could also in our actions do things like:
 * show the resulting binary size changes
 * show the resulting binary dep deltas
 etc.

Neat stuff indeed.

Reply:
I renamed this package `depcheck` here: https://github.com/knative/pkg/pull/1853

We can add other assertions about the dependency graph (e.g. `AssertOnlyDeps`), which I could see being very useful for our most widely used libraries.  However, generally I think this will be prohibitive to maintain for non-leaf libraries.